### DISCUSSION TITLE
How to customize Conventional Commits

### DISCUSSION THREAD

**Discussion by yipsmith** [renovatebot/renovate#11346] [Request Help] [answered]
Hi. I'm new to Renovate and is currently using a self-hosted instance of renovate 25.49 in bitbucket-server. Everything it working fine but I would like to update the commits that Renovate puts in to correspond to conventional commits. 

For example, every PR right now gets put in as the default chore(deps): update to {some version here}

I would like for the commits to be something like if Renovate is putting a PR for updating v1 -> v2, this is a major revision or a breaking change so I would like the commits to be something like this `fix(deps): update to v2 BREAKING CHANGE:`

Or  if its just a patch or minor upgrade say v1-> v1.1.0 then I want the commit to look something like this fix(deps): update to v1.1.0.

Is there any way I can achieve something like this?

It would also be cool, if I could put certain rules on the file Renovate is putting a PR for. Say for all files that are called example.tf or if they belong to a certain folder, then `fix(deps): update ` would suffice. Otherwise, I want Renovate to put commits like the conditions I've mentioned above. 

Thanks in advance.

**Comment by viceice** [comment, marked as answer]
See sample: https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/renovate.json

https://gitlab.com/renovate-bot/renovate-runner/-/merge_requests/200

**Comment by yipsmith** [comment]
Thanks! That was really helpful! Do you see anything wrong here? 


In my packageRules, I have these rules 

      {
            "description": "Trigger breaking release for major updates",
            "ignorePaths": ["examples/**"],
            "matchUpdateTypes": ["major"],
            "semanticCommitType": "feat",
            "commitBody": "BREAKING CHANGE:"
        },
        {
            "description": "Trigger non breaking release for major updates inside examples folder",
            "matchPaths": ["examples/**"],
            "matchUpdateTypes": ["major", "minor","patch"],
            "extends": [":semanticCommitTypeAll(chore)"]
        }


I can’t seem to get renovate to ignore my folder called examples. I want every dependency update put in by Renovate that belongs to the examples folder to be a chore. 

The result I’m getting is Renovate keeps marking the dependencies in the examples folder as breaking changes for the major updates.

**Reply by rarkins** [reply]
From this it would imply that there's something wrong with the ignorePaths. If you can replicate it in a public repo then we can take a further look
